discord_connector.py

```diff
@@ -0,0 +1,566 @@
+
+
+    def save_artifact_to_soar(self, artifact):
+        status, creation_message, artifact_id = self.save_artifact(artifact)
+        self.save_progress("creating artifact: status: {}, creation message: {}, artifact id {}"
+                           .format(status, creation_message, artifact_id))
+        return artifact_id
+
+
+    def parse_message(self, message, attachments, embeds):
+        return {
+            "message origin": {
+                "channel id": message.channel.id,
+                "channel name": message.channel.name,
+            },
+            "message data": {
+                "created at": str(message.created_at),
+                "edited at": str(message.edited_at) if message.edited_at is not None else "message was not edited",
+            },
+            "author data": {
+                "author id": message.author.id,
+                "author name": message.author.name,
+            },
+            "jump url": message.jump_url,
+            "flags": list(filter(lambda flag: getattr(message.flags, flag), MESSAGE_FLAGS)) or "no flags",
+            "attachments": attachments or "no attachments",
+            "embeds": embeds or "no embeds",
+            "content": message.content
+        }
+
+
+    def _handle_delete_message(self, param):
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        channel_id = param['channel_id']
+        message_id = param['message_id']
+
+        status = self.delete_message(channel_id, message_id, action_result)
+
+        summary = action_result.update_summary({})
+        summary['action result: '] = "Deleting message {} ended with {}".format(message_id,
+                                                                                "success" if status else "failure")
+        return action_result.set_status(phantom.APP_SUCCESS) if status else action_result.set_status(phantom.APP_ERROR)
+
+
+    def delete_message(self, channel_id, message_id, action_result):
+        status, message = self.fetch_message(channel_id, message_id, action_result)
+        if not status:
+            return status
+        status, result = self.run_in_loop(message.delete(), action_result, message="Unable to delete message.")
+        return status
+
+
+    async def _load_guild(self):
+        await self._client.login(self._token)
+        self._guild = await self._client.fetch_guild(self._guild_id)
+
+
+    def _handle_list_channels(self, param):
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        status, channels = self.run_in_loop(self._guild.fetch_channels(), action_result,
+                                            message="Cannot fetch channel from Discord.")
+
+        num_channels = 0
```

Comment:
I'd change that to `text_channels_count` it's a bit more descriptive, as we don't want to gather other type of channels

Reply:
Instead I used 
```summary['num_channels'] = sum(isinstance(channel, discord.TextChannel) for channel in channels)```
so we don't need to use any variable name